docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,428 @@
+
+export OIDC_ARN=$(jq -r .OpenIDConnectProviderArn ${PWD}/new-oidc//04-iam-identity-provider-object.json)
+
+echo ${OIDC_ARN}
+```
+
+## Backup existing state <a name="backup"></a>
+
+- Get Objects and existing tests
+
+```bash
+export CURRENT_PATH=$PWD/current-cluster
+mkdir $CURRENT_PATH
+oc get authentication -o yaml |tee -a $CURRENT_PATH/authentication.yaml
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}" \
+    | jq -r '.Credentials=""' \
+    | tee ${CURRENT_PATH}/identities.json
+```
+
+- Save the current IAM Roles
+
+```bash
+aws iam list-roles \
+    | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-openshift\"))" \
+    | tee ${CURRENT_PATH}/iam-roles.json
+```
+
+## Patch the cluster to use the new OIDC<a name="patch-cluster"></a>
+
+- Patch the trusted policy documents with the new OIDC URL
+
+```bash
+sed "s/${OIDC_BUCKET_HOST}/${CLOUDFRONT_HOST}/g" ${CURRENT_PATH}/iam-roles.json \
+    | tee iam-roles-new.json
+```
+
+- Patch the IAM Roles Trusted Policy documents
+
+> NOTE 1: from here, the cluster will lose access to the integrated components (machine-api, image registry, CSI, ...)
+
+> NOTE 2: The script below should be run carefully, it was created to show the current and desired policies. If you find anything that does not match the expected changes, abort it immediately.
+
+> Helper [`aws iam get-role`](https://docs.aws.amazon.com/cli/latest/reference/iam/get-role.html)
+
+> Helper [`aws iam update-assume-role-policy`](https://docs.aws.amazon.com/cli/latest/reference/iam/update-assume-role-policy.html)
+
+```bash
+for ROLE_NAME in $(jq -r .RoleName iam-roles-new.json);
+do
+    echo -e ">>>>>\n#> (1) CURRENT IAM Role \"$ROLE_NAME\":";
+    aws iam get-role --role-name $ROLE_NAME | jq .Role;
+
+    echo -e "\n#> (2) NEW IAM Role \"$ROLE_NAME\" AssumeRolePolicyDocument:";
+    jq -r ". | select(.RoleName == \"$ROLE_NAME\").AssumeRolePolicyDocument" iam-roles-new.json \
+        | tee ${PWD}/iam-roles-new-$ROLE_NAME.json
+    
+    read -p "ATTENTION: The AssumeRolePolicyDocument for IAM Role(1) will be patched to the value of (2). Do you want to continue? [y/n]: " answer
+    if [ -z "$answer" ] || [ "$answer" != "y" ]
+    then
+        echo "answer[$answer]. Canceling the operation.";
+        break
+    fi
+    echo "Patching..."
+    aws iam update-assume-role-policy \
+        --role-name $ROLE_NAME \
+        --policy-document file:///${PWD}/iam-roles-new-$ROLE_NAME.json
+    echo "Done! Return code=$?"
+done
+```
+
+- Patch the issuer URL to the new OIDC URL on the Authentication object:
+
+```bash
+oc patch authentication cluster \
+    --type=merge \
+    -p "{\"spec\":{\"serviceAccountIssuer\":\"https://${CLOUDFRONT_HOST}\"}}"
+```
+
+- Wait for the kube-apiserver rollout
+
+> Wait to clean the `PROGRESSING=TRUE`. It could take some minutes to start, and several to finish.
```

Comment:
"and several ... to finish",seems the unit is missing. In my testing, it was finished in < 10 minutes